accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,305 @@
+
+These collected data are stored into a report table with a status field.
+The status can be:
+
+- pending
+- failed
+- succeeded
+
+The initial status is `pending` which indicate that the verification has not yet happen.
+
+Writing the state result into the report table could be generalized by taking the state IDs as keys and the returned result as values.
+Putting this into a map and store it as Json into a data column.
+
+We also have a mapping between environment type and expected verification/results.
+At the end of step one all the expected result checks are created in the result table with the status `pending`.
+
+Environment Types and Result Types should be ENUM types in Java.
+Also the expected mappings which environment type require which result type will life in Java.
+
+
+### Step two
+The second step is a process which run in a dedicated environment like a container which verify the result.
+
+Such an attestation container is responsible for one or more result types.
+The attestation container search regular for pending attestations of its result type.
+It read the data from the report table, performs all the needed checks and update the result table with the outcome
+and set the status to `succeeded` or `failed`.
+
+The overall status of the report depends on the status of all required checks. A database trigger can fire on every change of
+the status in the result table and calculate the overall status. When this is different to the stored status
+in the report table, the trigger can update it and set it to `succeeded` of `failed`.
+
+
+## Database Design
+
+The table for the configuration should be a one-to-one relation to `rhnServer`.
+It has a boolean for attestation is enabled and an env_type column for the environment type of that system.
+
+```sql
+CREATE TABLE suseServerCoCoAttestationConfig
+(
+        server_id NUMERIC NOT NULL
+                    CONSTRAINT suse_srvcocoatt_cnf_sid_pk PRIMARY KEY,
+        CONSTRAINT suse_srvcocoatt_cnf_sid_fk FOREIGN KEY (server_id) REFERENCES rhnServer (id),
+        enabled   BOOLEAN NOT NULL DEFAULT FALSE,
+        env_type  NUMERIC NULL
+);
+```
+
+The table for the report should be a many-to-one relation to hold a history of the attestations.
+
+We have 2 data columns. The `in_data` column is used for input data we need to provide via
+state.apply to the VM. The input data should be added as pillar data for that minion.
+
+The `out_data` column contains the results coming back from the state.apply.
+These data columns are of type JSONB to hold any kind of data in machine readable form.
+As different environment types perform different checks, multiple data structures are expected.
+
+The status is calculated from the different results. It should be set to `succeeded` when all
+results have the status `succeeded` as well. As soon as one result is set to `failed` the
+overall status of the report should also show `failed`.
```

Comment:
Same as above. Yes, overall status will be failed but still would be good show what tests were and what failed.

Reply:
I explained the details view in the UI a bit better